contrib/tut/src/mill/contrib/tut/TutModule.scala

```diff
@@ -6,23 +6,96 @@ import coursier.MavenRepository
+  *
+  * {{{
+  * build.sc
+  * example/
+  *     src/
+  *     tut/
+  *     resources/
+  * }}}
+  *
+  * In order to compile documentation we can execute the `tut` task in the module:
+  *
+  * {{{
+  * sh> mill example.tut
+  * }}}
+  */
 trait TutModule extends ScalaModule {
+  /**
+    * This task determines where documentation files must be placed in order to be compiled with Tut. By default this is the `tut` folder at the root of the module.
+    */
   def tutSourceDirectory = T.sources { millSourcePath / 'tut }
+
+  /**
+    * A task which determines where the compiled documentation files will be placed. By default this is simply the Mill build's output folder for this task,
+    * but this can be reconfigured so that documentation goes to the root of the module (e.g. `millSourcePath`) or to a dedicated folder (e.g. `millSourcePath / 'docs`)
+    */
   def tutTargetDirectory: T[Path] = T { T.ctx().dest }
+
+  /**
+    * A task which determines what classpath is used when compiling documentation. By default this is configured to use the same inputs as the [[mill.contrib.tut.TutModule#runClasspath]],
+    * except for using [[mill.contrib.tut.TutModule#tutIvyDeps]] rather than the module's [[mill.contrib.tut.TutModule#runIvyDeps]].
+    */
   def tutClasspath: T[Agg[PathRef]] = T {
     // Same as runClasspath but with tut added to ivyDeps from the start
-    // This prevents duplicate copies of scala-library ending up on the classpath
+    // This prevents duplicate, differently versioned copies of scala-library ending up on the classpath which can happen when resolving separately
     transitiveLocalClasspath() ++
     resources() ++
     localClasspath() ++
     unmanagedClasspath() ++
     tutIvyDeps()
   }
+
+  /**
+    * A task which determines the scalac plugins which will be used when compiling code examples with Tut. The default is to use the [[mill.contrib.tut.TutModule#scalacPluginIvyDeps]] for the module.
+    */
   def tutScalacPluginIvyDeps: T[Agg[Dep]] = scalacPluginIvyDeps()
+
+  /**
+    * A [[scala.util.matching.Regex]] task which will be used to determine which files should be compiled with tut. The default pattern is as follows: `.*\.(md|markdown|txt|htm|html)`.
+    */
   def tutNameFilter: T[Regex] = T { """.*\.(md|markdown|txt|htm|html)""".r }
-  def tutScalacOptions: T[Seq[String]] = scalacOptions()
+
+  /**
+    * The scalac options which will be used when compiling code examples with Tut. The default is to use the [[mill.contrib.tut.TutModule#scalacOptions]] for the module,
+    * but filtering out options which are problematic in the REPL, e.g. `-Xfatal-warnings`, `-Ywarn-unused-imports`.
+    */
+  def tutScalacOptions: T[Seq[String]] =
+    scalacOptions().filterNot(Set(
+      "-Ywarn-unused:imports",
+      "-Ywarn-unused-import",
+      "-Ywarn-dead-code",
+      "-Xfatal-warnings"
+    ))
+
+  /**
+    * The version of Tut to use.
+    */
   def tutVersion: T[String] = "0.6.7"
```

Comment:
To follow the same pattern as in other mill modules (e.g. ScalaModule), we should not hard code any versions. Hence this field should be mandatory.

Reply:
Yep, seems reasonable

Reply:
OK, the latest commit removes the default version here